vm/src/obj/objbool.rs

```diff
@@ -150,14 +150,17 @@ impl PyBool {
     }
 
     #[pyslot]
-    fn tp_new(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-        arg_check!(
-            vm,
-            args,
-            required = [(_zelf, Some(vm.ctx.type_type()))],
-            optional = [(val, None)]
-        );
-        let value = match val {
+    fn tp_new(vm: &VirtualMachine, mut args: PyFuncArgs) -> PyResult {
```

Comment:
```suggestion
fn tp_new(zelf: PyObjectRef, value: PyObjectRef, vm: &VirtualMachine) -> PyResult
```

Unless there are keyword arguments, this form is more easy for most of case. In this way, you don't need to unpack zelf and value from args yourself.

Reply:
Is there another form I could use when I need to accept a zero-arguments case?

Reply:
`(args..., vm: &VirtualMachine) -> ReturnType` form work for everything.

These examples are actual RustPython code for zero-argument:
```rust
fn builtin_globals(vm: &VirtualMachine) -> PyResult<PyDictRef> { }
fn repr(self, vm: &VirtualMachine) -> PyResult<String> { }
```

If you are looking for optional arguments that would allow zero-arguments for users, check for `OptionalArg` for positional-only argument
```rust
fn sys_exit(code: OptionalArg<PyObjectRef>, vm: &VirtualMachine) -> PyResult
```
and search for `FromArgs` for keyword arguments.

Reply:
I tried using `OptionalArg` for `tp_new`, but it seems that it's not suitable for `tp_new` at the moment, since it allows keyword arguments to be passed as well. For example, when used in this form:

`fn tp_new(zelf: PyObjectRef, x: OptionalArg<PyObjectRef>, vm: &VirtualMachine)`

It fails the test `test_keyword_args` in `test_bool.py`.

Reply:
Oh, that's fine, just remove the `unexpectedFailure` annotation from `test_keyword_args`.

Reply:
@coolreader18 I'm not sure if I used `OptionalArg` correctly, but I've added the related commits!